Draft start of episode, departing from Sarah's repo (#53)
carpentries-incubator/building-websites-with-jekyll-and-github-or-gitlab: _episodes/02-markdown.md

- put directory under version control with git
  - link to SWC git-novice

# Markdown
Markdown is a lightweight markup language, i.e. a convention for adding
style information to textual content.
As the name Markdown indicates, the syntax elements of this language
are shut *down* to a minimum.
Having a rather minimalistic syntax, text formatted in Markdown is comparably

> well readable.

Reviewer: ```suggestion
readable.
```

This might be one reason for Markdown having become the language of choice
for formated user input on websites like for example
- [Stack Exchange](https://stackexchange.com/)
- [GitHub](https://github.com/)
- [GitLab](https://about.gitlab.com/).

# Where to start writing Markdown?
A lot of tools for rendering Markdown source code exist.
Rendering is the process of generating a nice view of the content
using the style information included in the source text.
Chances are high, your editor can do this.

> As we are working towards authoring websites using Jeckyll and GitHub pages,

Reviewer: ```suggestion
As we are working towards authoring websites using Jekyll and GitHub pages,
```

we will use GitHub straight away for learning the basics of Markdown.
The GitHub project you created in the last episode contains a file `README.md`.

The picture below shows the projects default view.
This view includes a rendered view of the content inside the file `README.md`.

> Your project should look quite similar except for the red circle around the pencil symbol.

Reviewer: Love the red circle!  I've noticed this pencil icon is not intuitive for most people to find.

> ![Warning: You should see a picture here displaying: Github repository for the group website](../fig/group_website_repo_edit.png)

Reviewer: I've not seen alt text written like this before, is it accessible?  My understanding is that you want to describe want is in the image and avoid extra words like "Picture of" etc.  Examples [here](https://accessibility.huit.harvard.edu/describe-content-images)

Reviewer 2: maybe "On the homepage of a GitHub repository, the edit button is positioned to the top right of the README preview." ?

Author: You are totally right, thanks for pointing towards the standard and giving a perfect example.

> You can click on that pencil symbol to open an edit view of your projects README file.

Reviewer: ```suggestion
You can click on that pencil symbol to open an editing interface of your project's `README.md` file.
```

Reviewer 2: I would prefer "editing interface" to "edit view" as I think that makes a clearer distinction between the editing screen and the raw view.

Reviewer: Edited suggestion to included above

> This looks like

Reviewer: Maybe want to add what it looks like in word form here, in case folks can't see the image.

```suggestion
Once you've clicked the edit button, GitHub will open that file in its web text editor.
```

Author: I'll reuse the wording `editing interface` and your suggestion!

![Warning: You should see a picture here displaying: Editing the group websites README file](../fig/group_website_readme_edit.png)

> displaying the source of your projects README file.

Reviewer: instead of my suggestion above, you could move this line to finish before the image.

You can change this content and have a look at the rendered view by clicking the `Preview changes` tab.

> This looks like

Reviewer: I'm thinking here you should have them change something about the readme, and then see the preview.

Like... Let's type `**bold**` and see what happens when we preview it using the preview tab.

Author: I make a suggestion step by step, so first talking about Preview-tab and then have the learners write formatted markdown the first time

![Warning: You should see a picture here displaying: Rendering the group websites README file](../fig/group_website_readme_render.png)

If you like to permanently save the content to the file `README.md`, scroll down a bit and commit your changes.
After having changed something, the commit menu looks like this

> ![Warning: You should see a picture here displaying: Commit changes to the README file](../fig/group_website_readme_commit.png)

Reviewer: Maybe a callout here with a little info about writing a commit message?

Reviewer 2: +1 and add a link to the relevant section of the git-novice lesson for those who need a refresher/want to read more about commit messages.

>

> > Try to reproduce the source code of the view which is rendered below this line:

Reviewer: Maybe we need to give them a cheat sheet of markdown so they can find the syntax for this exercise?

Reviewer 2: I like https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet
